crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -3544,50 +3496,28 @@ impl AuthorityPerEpochStore {
     fn process_consensus_transaction_shared_object_versions(
         &self,
         cache_reader: &dyn ObjectCacheRead,
-        non_randomness_transactions: &[VerifiedExecutableTransaction],
-        randomness_transactions: &[VerifiedExecutableTransaction],
-        randomness_round: Option<RandomnessRound>,
+        non_randomness_transactions: &[Schedulable],
+        randomness_transactions: &[Schedulable],
         cancelled_txns: &BTreeMap<TransactionDigest, CancelConsensusCertificateReason>,
         output: &mut ConsensusCommitOutput,
-    ) -> SuiResult {
-        // If randomness_round is set, we know that eventually there will be a randomness state update transaction.
-        // We create a placeholder transaction so that the SharedObjVerManager
-        // can update the version of the randomness state object and use that version for randomness transactions.
-        let randomness_state_update = randomness_round.map(|round| {
-            VerifiedExecutableTransaction::new_system(
-                VerifiedTransaction::new_randomness_state_update(
-                    self.epoch(),
-                    round,
-                    // This is placeholder bytes, since this transaction does not exist yet.
-                    vec![],
-                    self.epoch_start_config()
-                        .randomness_obj_initial_shared_version()
-                        .expect("randomness obj initial shared version should be set if randomness_round is set"),
-                ),
-                self.epoch(),
-            )
-        });
-        let all_certs = non_randomness_transactions
+    ) -> SuiResult<AssignedTxAndVersions> {
+        let all_certs: Vec<_> = non_randomness_transactions
             .iter()
-            // randomness_state_update must be before randomness_transactions to make sure the version
-            // of the randomness state object is updated before it is used in randomness transactions.
-            .chain(randomness_state_update.iter())
-            .chain(randomness_transactions.iter());
+            .chain(randomness_transactions.iter())
+            .collect();
+
         let ConsensusSharedObjVerAssignment {
             shared_input_next_versions,
             assigned_versions,
         } = SharedObjVerManager::assign_versions_from_consensus(
             self,
             cache_reader,
-            all_certs,
+            all_certs.into_iter(),
```

Comment:
why the extra `all_certs` `Vec`? would it work to pass in just the chained iters here (without collect)?

Reply:
yes, works!